CMakeLists.txt

```diff
@@ -127,6 +128,7 @@ if(NOT MSVC)
 	enable_cxx_compiler_flag_if_supported("-Wredundant-decls")
 	enable_cxx_compiler_flag_if_supported("-Wshadow")
 	enable_cxx_compiler_flag_if_supported("-Woverloaded-virtual")
+	enable_cxx_compiler_flag_if_supported("-Werror")
```

Comment:
As a developer, I am very supportive of building with `-Werror` locally and on CI, and I applaud any effort to make this possible.  But I'd caution against enabling `-Werror` by default for everyone who builds Aer.  While the code might build without warnings on specific versions of specific compiler(s), it is still quite possible for a different compiler (or even a newer release of the same compiler, e.g. gcc) to trigger a new warning.  Is it really better for users to error out rather than to continue with the build when this happens?  In my experience, supporting users through the resulting (and ongoing) fallout can be a major headache.